### ISSUE TITLE
Add support for Implied Roles.

### ISSUE THREAD

**Issue by xuxant** [gophercloud/gophercloud#2225]
Adding support for implied roles in Gophercloud.

**Comment by xuxant** [comment]
@jtopjian Can you please have a look at the above mention APIS? Can we implement them?

**Comment by jtopjian** [comment]
@xuxant Yes, just make sure to follow the [contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial). Please let me know if you have any questions about it - happy to help.
